rpcd/playbooks/roles/elasticsearch/tasks/upgrade_elasticsearch_pre.yml

```diff
@@ -0,0 +1,200 @@
+    port: 9200
+    state: started
+    timeout: 300
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Drop log_test1 template
+  shell: 'curl -sk -XDELETE {{ internal_lb_vip_address }}:{{ elasticsearch_http_port }}/_template/log_test1'
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+    - skip_ansible_lint
+
+- name: Place legacy liberty template script
+  template:
+    src: legacy-mapping.sh.j2
+    dest: /opt/legacy-mapping.sh
+    owner: root
+    group: root
+    mode: 0755
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Install legacy liberty template
+  shell: '/opt/legacy-mapping.sh'
+  register: legacy_template_result
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Remove legacy liberty template script
+  file:
+    name: '/opt/legacy-mapping.sh'
+    state: absent
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Install Reindexing Plugin
+  command: ./plugin -install {{ item }}
+  args:
+    chdir: /usr/share/elasticsearch/bin
+    creates: /usr/share/elasticsearch/plugins/reindexing
+  with_items:
+    - org.codelibs/elasticsearch-reindexing/1.7.0
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Install reindexing wrapper script
+  copy:
+    src: reindex-liberty.py
+    dest: /opt/reindex-liberty.py
+    owner: root
+    group: root
+    mode: 0755
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-reindex
+    - reindex-wrapper
+
+- name: Reindex legacy elasticsearch indices
+  shell: /opt/reindex-liberty.py --host "{{ hostvars[groups['elasticsearch_container'][0]]['container_address'] }}" --reindex
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-reindex
+
+- name: Monitor reindexing process
+  shell: /opt/reindex-liberty.py --host "{{ hostvars[groups['elasticsearch_container'][0]]['container_address'] }}" --monitor --continuous
+  async: 259200
+  poll: 5
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-reindex
+
+- name: Drop legacy elasticsearch indeices
+  shell: /opt/reindex-liberty.py --host "{{ hostvars[groups['elasticsearch_container'][0]]['container_address'] }}" --drop
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-reindex
+
+- name: Remove Reindexing Plugin
+  command: ./plugin remove {{ item }}
```

Comment:
I might of mis-lead you earlier. The usage of this executable is different across elasticsearch versions. Before an upgrade, it is:
```
root@cantu-testup-logging6_elasticsearch_container-3d27670d:/usr/share/elasticsearch/bin# ./plugin -h
Usage:
    -u, --url     [plugin location]   : Set exact URL to download the plugin from
    -i, --install [plugin name]       : Downloads and installs listed plugins [*]
    -t, --timeout [duration]          : Timeout setting: 30s, 1m, 1h... (infinite by default)
    -r, --remove  [plugin name]       : Removes listed plugins
    -l, --list                        : List installed plugins
    -v, --verbose                     : Prints verbose messages
    -s, --silent                      : Run in silent mode
    -h, --help                        : Prints this help message
```
AFTER an upgrade, it is:
```
root@cantu-testup-logging4-elasticsearch-container-be510627:/usr/share/elasticsearch/bin# ./plugin -h

NAME

    plugin - Manages plugins

SYNOPSIS

    plugin <command>

DESCRIPTION

    Manage plugins

COMMANDS

    install    Install a plugin

    remove     Remove a plugin

    list       List installed plugins

NOTES

    [*] For usage help on specific commands please type "plugin <command> -h"
```
Since this will be executed before elasticsearch is upgraded, it should be ``./plugin --remove [plugin name]``, right?